File: java/com/kkucharski/rose/RxMath.java
package com.kkucharski.rose;

class RxMath {
    private static final float EPSILON = 0.000001f;

    static boolean approximately(float a, float b)
    {
        float absA = Math.abs(a);
        float absB = Math.abs(b);
        float diff = Math.abs(a - b);

        if (a == b) {
            return true;
        }

        if (a == 0.0f || b == 0 || diff < EPSILON) {
            return diff < EPSILON;
        }

        return diff / (absA + absB) < EPSILON;
    }
}

File: c#/RxMath.cs
using System;

namespace Rose
{
    public static class RxMath
    {
        private const float EPSILON = float.Epsilon;

        public static bool Approximately(float a, float b)
        {
            var absA = Math.Abs(a);
            var absB = Math.Abs(b);
            var diff = Math.Abs(a - b);

            if (a == b)
            {
                return true;
            }
            if (a == 0.0f || b == 0 || diff < EPSILON)
            {
                return diff < EPSILON;
            }

            return diff / (absA + absB) < EPSILON;
        }

        public static float Abs(float f)
        {
            return Math.Abs(f);
        }
    }
}